allennlp/models/encoder_decoders/wikitables/wikitables_semantic_parser.py

```diff
@@ -136,12 +155,66 @@ def __init__(self,
                                                    mixture_feedforward=mixture_feedforward,
                                                    dropout=dropout)
 
+        self._executor_process: subprocess.Popen = None
+        self._create_sempre_executor()
+
+    def _create_sempre_executor(self) -> None:
+        """
+        Creates a server running SEMPRE that we can send logical forms to for evaluation.  This
+        uses inter-process communication, because SEMPRE is java code.  We also need to be careful
+        to clean up the process when our program exits.
+        """
+        if self._executor_process:
+            return
+        os.makedirs(SEMPRE_DIR, exist_ok=True)
+        abbreviations_path = os.path.join(SEMPRE_DIR, 'abbreviations.tsv')
```

Comment:
These should either use the allennlp cache or raise errors when they aren't present (the EVALB metric just raises an error - it seems like that would be a good fit here too, because if I was training a model, I would want to know how it was being evaluated rather than it just magically happen for me).